network/crl.go

```diff
@@ -346,6 +347,35 @@ func (v DefaultCRLVerifier) getCachedOrFetch(url string, allowLocal bool, issuer
 // Returns `nil` if certificate was not cound in CRL, returns error if it was there
 // or if there was unknown Object ID in revoked certificate extensions
 func checkCertWithCRL(cert *x509.Certificate, crl *pkix.CertificateList) error {
+	for _, extension := range crl.TBSCertList.Extensions {
+		// For CRL v2 (RFC 5280 section 5.2), CRL issuers are REQUIRED to include
+		// the authority key identifier (Section 5.2.1) and the CRL number (Section 5.2.3).
+		// This means we also have to handle these extensions.
+		switch extension.Id.String() {
+		case "2.5.29.35":
+			// section 5.2.1 (4.2.1.1), id-ce-authorityKeyIdentifier
+
+		case "2.5.29.18":
+			// section 5.2.2 (4.2.1.7), id-ce-issuerAltName
+
+		case "2.5.29.20":
+			// section 5.2.3, id-ce-cRLNumber
```

Comment:
Don't we need to do something here? At least a log line should be nice. Or are those extensions handled by Go's x509?

Reply:
> Or are those extensions handled by Go's x509?

Nope, these are not handled. All the stuff related to CRL is like "if you need it, take these simple structs from crypto/x509/pkix and implement needed functions by yourself"

> At least a log line should be nice.

okay, gonna add some logs for all these `case`s. Debug or something more "serious"?

Reply:
Thanks for clarifications.

I'm just not sure why this function is necessary. It seems to be filtering out CRLs that Acra 'does not handle' (because it does not recognize the extensions). And yet, those extensions that it does recognize don't seem to be used in any way. The only side effect here are those debug-level logs about unrecognized extensions, and a possible failure if there is some critical extension which is not listed here. And if it were listed here, the code would just do nothing.

So I wonder, what's the point here. Discourage people from putting weird extensions?

Reply:
The point was to create some skeleton for further extension of cert verifier using CRL. I guess you wanna way "better to not have these "empty switches" at all or have them doing what they have to do"?

Reply:
Well, yeah, something like that. Though, I did not realize that you intend them as a scaffolding for future development. I'm sorry. In that case I guess a TODO comment will be better than logs. Just to leave a message to whoever will be reading this code.